cli/Dockerfile

```diff
@@ -17,11 +17,10 @@ RUN export DEBIAN_FRONTEND=noninteractive \
      # az cli
      && curl -sSL https://aka.ms/InstallAzureCLIDeb | bash \
      # nodejs
-     && curl -fsSL https://deb.nodesource.com/setup_18.x | bash - \
-     && apt-get install -y nodejs \
+     #&& curl -fsSL https://deb.nodesource.com/setup_18.x | bash - \
+     && apt-get install -y nodejs gcc g++ make npm\
```

Comment:
Why are `gcc`, `g++`, and `make` needed?

Reply:
No idea, @vhvb1989 can you provide some context?

Reply:
This is after removing: 

`&& curl -fsSL https://deb.nodesource.com/setup_18.x | bash - \`

We were using https://deb.nodesource.com/setup_18.x to install node, but it is not deprecated.

I saw a message at the end of the script about installing c/c++ for building native:

![image](https://github.com/Azure/azure-dev/assets/24213737/25385057-a9a6-45fa-8924-708b00de1f5e)

Reply:
I'll update by using the installation instructions from https://github.com/nodesource/distributions

Reply:
updated as it should now.
Thank you @danieljurek